src/js/yeldaWebchatInjector.js

```diff
@@ -620,23 +582,98 @@ class YeldaChat {
 
     return isFound
   }
+
+
+  /**
+   * Update assistantImage with assistant settings from backend if any
+   * @param {Object} data { data.assistantUrl, data.assistantId }
+   * @return {Promise}
+  */
+  getAssistantSettings(data) {
+    return new Promise(resolve => {
+      try {
+        const xhr = new XMLHttpRequest();
+        const url= `${data.assistantUrl}/assistants/${data.assistantId}/chatBubble/${data.locale}`
+
+        xhr.open("GET", url);
+        xhr.send();
+
+        // Bind and call are necessary to pass the "this" to the callback function
+        xhr.onreadystatechange = (function () {
+          if(xhr.readyState === 4) {
+            this.webchatSettings = xhr.responseText ? JSON.parse(xhr.responseText).data : null
+            resolve()
+          }
+        }).bind(this)
+      } catch (e) { // when json.parse fails or xhr onerror catch will be called
+        this.webchatSettings = null
+        resolve()
+      }
+    })
+  }
+
```

Comment:
As suggested in the previous review comment https://github.com/Yeldaai/chat-plugin/pull/85#discussion_r589816453
Can we 
- make getAssistantSettings independant from `this` 
- reduce its duty to 1 thing = getting data (and assignment will be handled outside)
- reject if it fail (and catch will be done outside, where it should be)

ex
```js
  // Get assistant settings from backend
 this.getAssistantSettings(data)
   .then(webchatSettings => {
     this.webchatSettings = webchatSettings
     resolve()
   }.catch(() => {
       this.webchatSettings = null
       resolve()  
   }
````